### PR TITLE
[google_sign_in_web] Ensure web plugin throws PlatformExceptions

### PR DESCRIPTION
**Pull request by ditman**
## Description

The web version of the plugin wasn't very good at throwing exceptions:

* Errors on `init()` weren't being unpacked correctly in `release` or `profile` mode, because of a missing JS-interop class. This prevented the code from throwing a proper PlatformException.
* Errors on `signIn()` were being rethrown as JS Objects, instead of wrapping them correctly in a PlatformException.

This PR ensures that the methods above throw the appropriate exception every time, regardless of the environment (dev vs release).

This PR also aligns the fields of the PlatformException with those coming from the mobile side, so:

* `PlatformException.code` is now what the Google Sign In SDK calls `"error"`.
* `.message` is now what the Google Sign In SDK calls `"details"` (when available).
* `.details` is `'google_sign_in'`

See multiple "Error Codes" here: https://developers.google.com/identity/sign-in/web/reference#error_codes

Finally, the tests on the package have been migrated to run with `flutter drive` + `chromedriver` instead of other options.

## Related Issues

* https://github.com/flutter/flutter/issues/52338
* https://github.com/flutter/flutter/issues/59145

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Comment by ditman** [comment]
> Can we test this?

@cyanglaz I need to migrate the tests to e2e first, but I can shave that yak for this PR, yes.

(Main goal of publishing this early was for people affected by the bugs above to try this branch and see if their apps behave better)

**Comment by ditman** [comment]
@cyanglaz I've migrated the tests to run as `flutter drive` (instead of `flutter test --platform chrome`), and added a few tests for the new behavior. PTAL!

File: packages/google_sign_in/google_sign_in_web/lib/google_sign_in_web.dart [hunk 1]
```diff
@@ -128,8 +128,15 @@ class GoogleSignInPlugin extends GoogleSignInPlatform {
...
     await initialized;
-
-    return gapiUserToPluginUserData(await auth2.getAuthInstance().signIn());
+    try {
+      return gapiUserToPluginUserData(await auth2.getAuthInstance().signIn());
+    } on auth2.GoogleAuthSignInError catch (reason) {
+      throw PlatformException(
+        code: reason.error,
+        message: 'Exception raised from GoogleAuth.signIn()',
+        details: 'google_sign_in',
```
Comment (cyanglaz) [inline packages/google_sign_in/google_sign_in_web/lib/google_sign_in_web.dart:137]: `google_sign_in` as the details doesn't seem to be necessary. Maybe change the message to 
`google_sign_in: Exception raised from GoogleAuth.signIn()` 
And for details, we can put more details if we have (stack trace etc)

Comment (ditman) [reply packages/google_sign_in/google_sign_in_web/lib/google_sign_in_web.dart:137]: The only guaranteed information we have from the JS SDK here is the ["error" property](https://developers.google.com/identity/sign-in/web/reference#googleauthsignin) that we're using as the code.

Maybe in `details` I can add the link to the JS SDK so people can understand what the error means? (the values are pretty self-explanatory, though)

Comment (ditman) [reply packages/google_sign_in/google_sign_in_web/lib/google_sign_in_web.dart]: Ended up adding the link!

File: packages/google_sign_in/google_sign_in_web/lib/google_sign_in_web.dart [hunk 2]
```diff
@@ -105,16 +105,16 @@ class GoogleSignInPlugin extends GoogleSignInPlatform {
...
       // onError
-      throw PlatformException(
-        code: 'google_sign_in',
-        message: reason.error,
-        details: reason.details,
-      );
+      isAuthInitialized.completeError(PlatformException(
+        code: reason.error,
+        message: reason.details,
+        details: 'google_sign_in',
```
Comment (cyanglaz) [inline packages/google_sign_in/google_sign_in_web/lib/google_sign_in_web.dart:113]: ditto.

Comment (ditman) [reply packages/google_sign_in/google_sign_in_web/lib/google_sign_in_web.dart:113]: :facepalm:

Comment (ditman) [reply packages/google_sign_in/google_sign_in_web/lib/google_sign_in_web.dart]: Done!

**Review by cyanglaz — APPROVED** [review]
LGTM

**Comment by ditman** [comment]
Published [v0.9.2](https://pub.dev/packages/google_sign_in_web).

**Pull request merged**